shogun-lib/src/main/java/de/terrestris/shogun/lib/controller/security/permission/BasePermissionController.java

```diff
@@ -753,6 +754,78 @@ public void deleteGroupClassPermissions(
         }
     }
 
+    @GetMapping("/{id}/permissions/public")
+    @ResponseStatus(HttpStatus.OK)
+    public Map<String, Boolean> isPublic(
+        @PathVariable("id") Long entityId
+    ) {
+        try {
+            Optional<S> entity = service.findOne(entityId);
+
+            if (entity.isEmpty()) {
+                throw new EntityNotFoundException(entityId, getGenericClassName(), messageSource);
+            }
+
+            return Map.of("public", publicInstancePermissionService.getPublic(entity.get()));
+        } catch (AccessDeniedException ade) {
+            throw new EntityAccessDeniedException(entityId, getGenericClassName(), messageSource);
```

Comment:
I'd suggest to log the stack traces and error messages here and in the following catch clauses to enable better debugging